ci: test and analyze

wizlif opened the pull request.

wizlif committed:
ci: test and analyze

Signed-off-by: Isaac Obella <wizlif@users.noreply.github.com>

wizlif committed:
fix: format errors

Signed-off-by: Isaac Obella <wizlif@users.noreply.github.com>

wizlif committed:
ci: add code coverage

Signed-off-by: Isaac Obella <wizlif@users.noreply.github.com>

Xazin commented on .github/workflows/test-analyze.yml:
```diff
@@ -0,0 +1,92 @@
+
+on:
+  push:
+    branches: [master, development]
```
```suggestion
    branches: [project/**, release/**]
```
We don't have development or master branch.

You can also just remove `push` and only have `pull_request`.

wizlif committed:
Update .github/workflows/test-analyze.yml

Co-authored-by: Mathias Mogensen <42929161+Xazin@users.noreply.github.com>

wizlif committed:
fix: only apply for push

wizlif committed:
core: improve code coverage filters

wizlif committed:
chore: auto search codecov files.

Instead of manually adding codecov files, CI can search and auto append the codecov files.

Signed-off-by: Isaac Obella <wizlif@users.noreply.github.com>

Xazin commented on .github/workflows/test-analyze.yml:
```diff
@@ -0,0 +1,98 @@
+
+on:
+  pull_request:
+    branches: [project/**, release/**]
```
Let's also add main for now until we have figured out how we will use release branches.

wizlif replied:
Sure

wizlif replied:
Btw what will the project/** branches be for ??

wizlif committed:
chore: add main branch

Signed-off-by: Isaac Obella <wizlif@users.noreply.github.com>
